### PR TITLE
Nav cleanup

### PR DESCRIPTION
**Pull request by KevinJJackson**
### Summary
@mhemesath @alex-bezek @Mahmud-Khan 

Added
- Icon support for navigation items.
- target attribute support for navigation items.

Changed
- Profile settings modal now opens as expected in safari tablet mode.
- Minor UI fixes.
- PropType consistency for links/paths.

Thanks for contributing to Terra. 
@cerner/terra

### CODE REVIEW

File: packages/terra-consumer-nav/src/components/modal/Modal.jsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (mhemesath) [inline packages/terra-consumer-nav/src/components/modal/Modal.jsx]: Why do we have our own modal nested in consumer-nav?

File: packages/terra-consumer-nav/src/components/nav-items/NavItem.scss [hunk 2]
```diff
@@ -21,6 +21,15 @@
     }
   }
 
+  .nav-item-icon svg {
```
Comment (mhemesath) [inline packages/terra-consumer-nav/src/components/nav-items/NavItem.scss]: this wont' suffice for font icons... will need to set a font-size to control those

File: packages/terra-consumer-nav/src/components/nav-logo/NavLogo.jsx [hunk 3]
```diff
@@ -27,12 +27,12 @@ const defaultProps = {
...
 
 const NavLogo = ({
-  path,
+  url,
   altText,
   isCard,
   ...customProps
 }) => {
-  const image = <img className={cx('img')} src={path} alt={altText} />;
+  const image = <img className={cx('img')} src={url} alt={altText} />;
```
Comment (mhemesath) [inline packages/terra-consumer-nav/src/components/nav-logo/NavLogo.jsx]: I'm actually wondering if we shuld just have the navLogo be a react component.... Then all the link, card, and other logic can be handled outside of the nav framework.

File: packages/terra-consumer-nav/CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,19 @@
 ChangeLog
 =========
 
+# 0.1.0-BETA.3 - (--)
```
Comment (alex-bezek) [inline packages/terra-consumer-nav/CHANGELOG.md]: can we make sure to add this date of whatever day you commit? Otherwise I have to keep updating it every time i release

File: packages/terra-consumer-nav/src/Nav.jsx [hunk 5]
```diff
@@ -16,9 +19,7 @@ const propTypes = {
   /**
    * An array of objects to be displayed as quick link options.
    */
-  quickLinks: PropTypes.arrayOf(PropTypes.shape({
-    text: PropTypes.string.isRequired,
-  })),
```
Comment (alex-bezek) [inline packages/terra-consumer-nav/src/Nav.jsx]: why was this removed?

File: packages/terra-consumer-nav/src/components/modal/Modal.jsx (hunk 1 above)
Comment (KevinJJackson) [inline packages/terra-consumer-nav/src/components/modal/Modal.jsx]: this is definitely feasible and makes sense to me. I can do that separately from this PR if that's better.

File: packages/terra-consumer-nav/CHANGELOG.md (hunk 4 above)
Comment (KevinJJackson) [inline packages/terra-consumer-nav/CHANGELOG.md]: yeah that's my bad, i thought the release date was supposed to go here since releases may not always be the same day as the commit. I'll update this

File: packages/terra-consumer-nav/src/Nav.jsx (hunk 5 above)
Comment (KevinJJackson) [inline packages/terra-consumer-nav/src/Nav.jsx]: it wasn't removed, just changed. I just removed the validation from the nav to leave the more specific validation to the component.

File: packages/terra-consumer-nav/src/components/modal/Modal.jsx (hunk 1 above)
Comment (alex-bezek) [inline packages/terra-consumer-nav/src/components/modal/Modal.jsx]: I think this is the master plan in the end (same with card and popup). This is here for now for speed

File: packages/terra-consumer-nav/src/NavPropShapes.js [hunk 6]
```diff
@@ -6,14 +6,18 @@ const navItemShape = {
    */
   text: PropTypes.string.isRequired,
   /**
-   * The path the profile link would redirect to.
+   * The path the link would redirect to.
    */
-  uri: PropTypes.string.isRequired,
+  url: PropTypes.string,
```
Comment (KevinJJackson) [inline packages/terra-consumer-nav/src/NavPropShapes.js]: oops, removed this when i was debugging a weird issue, will put it back, thanks for pointing it out!

File: packages/terra-consumer-nav/src/components/modal/Modal.jsx [hunk 7]
```diff
@@ -14,50 +15,44 @@ const propTypes = {
    */
   title: PropTypes.string,
   /**
-   * Content to be rendered in the body of the dialog box.
-   */
-  content: PropTypes.element,
```
Comment (alex-bezek) [inline packages/terra-consumer-nav/src/components/modal/Modal.jsx]: is most of this just an attempt to align us with the terra modal more closely?
https://github.com/cerner/terra-core/blob/master/packages/terra-modal/src/Modal.jsx

Comment (KevinJJackson) [inline packages/terra-consumer-nav/src/components/modal/Modal.jsx]: for the most part, figured some consistency would be nice. also removed some unneeded nested classes from the title.

File: packages/terra-consumer-nav/src/NavPropShapes.js (hunk 6 above)
Comment (Mahmud-Khan) [inline packages/terra-consumer-nav/src/NavPropShapes.js]: There may be a case where the url is not needed. Like clicking a navigation item expand a sub item like the following.
screen shot 2017-08-24 at 3 20 59 pm

File: packages/terra-consumer-nav/src/components/nav-logo/NavLogo.jsx (hunk 3 above)
Comment (alex-bezek) [inline packages/terra-consumer-nav/src/components/nav-logo/NavLogo.jsx]: Are you saying remove this completely and let the consumer provide this as a prop to `Nav` or to just export this module for use by the consumer?

File: packages/terra-consumer-nav/src/NavPropShapes.js (hunk 6 above)
Comment (KevinJJackson) [inline packages/terra-consumer-nav/src/NavPropShapes.js]: ah you are correct here, we were previously bypassing the required field by simply passing `url: '',` into the component. 
example: https://github.com/cerner/terra-consumer/blob/master/packages/terra-consumer-nav/tests/nightwatch/DefaultNav.jsx#L37

File: packages/terra-consumer-nav/CHANGELOG.md (hunk 4 above)
Comment (alex-bezek) [inline packages/terra-consumer-nav/CHANGELOG.md]: I think you are technically correct. But i'm not sure the date matters that much. I'll work with @bjankord to see what his process is to make things easiest

File: packages/terra-consumer-nav/src/Nav.jsx (hunk 5 above)
Comment (alex-bezek) [inline packages/terra-consumer-nav/src/Nav.jsx]: i added just the text validation here because text is used below for the key. The idea is if a component takes props and uses some value off them, it should declare the type for that prop's value.

File: packages/terra-consumer-nav/src/NavPropShapes.js (hunk 6 above)
Comment (alex-bezek) [inline packages/terra-consumer-nav/src/NavPropShapes.js]: thats a good point @Mahmud-Khan . Not required makes sense.

File: packages/terra-consumer-nav/src/Nav.jsx (hunk 5 above)
Comment (KevinJJackson) [reply packages/terra-consumer-nav/src/Nav.jsx]: right, makes sense, will revert this change
